### ISSUE TITLE
grpc interop: `status_code_and_message` fails

### ISSUE THREAD

**Issue by stevej** [linkerd/linkerd#985]
```/Users/stevej% ./go-grpc-interop-client -use_tls=false  -server_port=60001 -test_case=status_code_and_message
2017/01/25 13:33:07 &{0xc420184500}.UnaryCall(_, response_status:<code:2 message:"test status message" > ) = _, <nil>, want _, rpc error: code = 2 desc = test status message```

Server output
```Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------OUTBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] SETTINGS: ack=false, settings={}
------------------------------------
Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] SETTINGS: ack=false, settings={}
------------------------------------
Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------OUTBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] SETTINGS: ack=true
------------------------------------
Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] WINDOW_UPDATE: streamId=0, windowSizeIncrement=983025
------------------------------------
Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] HEADERS: streamId=1, headers=DefaultHttp2Headers[:method: POST, :scheme: http, :path: /grpc.testing.TestService/UnaryCall, :authority: 127.0.0.1, content-type: application/grpc, user-agent: grpc-go/1.0, te: trailers], padding=0, endStream=false
------------------------------------
Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] DATA: streamId=1, padding=0, endStream=true, length=30, bytes=00000000193a17080212137465737420737461747573206d657373616765
------------------------------------
Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------OUTBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] HEADERS: streamId=1, headers=DefaultHttp2Headers[:status: 200], streamDependency=0, weight=16, exclusive=false, padding=0, endStream=false
------------------------------------
Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------OUTBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] DATA: streamId=1, padding=0, endStream=false, length=5, bytes=0000000000
------------------------------------
Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------OUTBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] HEADERS: streamId=1, headers=DefaultHttp2Headers[grpc-status: 0], streamDependency=0, weight=16, exclusive=false, padding=0, endStream=true
------------------------------------
Jan 25, 2017 1:33:46 PM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xa3cb63e5, L:/127.0.0.1:60001 - R:/127.0.0.1:61454] SETTINGS: ack=true
------------------------------------
Jan 25, 2017 1:33:46 PM com.twitter.finagle.buoyant.h2.netty4.Netty4DispatcherBase$$anonfun$loop$lzycompute$1$1 apply
ERROR: [S L:/127.0.0.1:60001 R:/127.0.0.1:61454] dispatcher failed
com.twitter.finagle.ChannelClosedException: ChannelException at remote address: /127.0.0.1:61454. Remote Info: Not Available
```

**Comment by olix0r** [comment]
We didn't support status messages before. Now that we do, the test server needs to be changed to support reading the response_status field on requests. https://github.com/linkerd/linkerd/commit/2e61fb272f940cec6391562f82af3ed8414becf8

**Comment by stevej** [comment]
I think we should leave this open until the interop test passes.

**Comment by olix0r** [comment]
fixed in #1024
